src/main/java/gg/solarmc/solarcredits/command/commands/SpendCommand.java

```diff
@@ -0,0 +1,123 @@
+
+public record SpendCommand(SolarCredit plugin, String X_TEBEX_SECRET) implements CreditSubCommand {
+    @Override
+    public boolean execute(CommandSender sender, String[] args, CommandHelper helper) {
+        if (sender instanceof Player player) {
+            if (args.length >= 1) {
+                final String amountString = args[0];
+                double amount = helper.getValidNumber(amountString);
+                final BigDecimal amountInDecimal = BigDecimal.valueOf(amount);
+
+                if (amount != -1) {
+                    ItemStack giftCard = new ItemStack(Material.MAP);
+                    ItemMeta giftMeta = giftCard.getItemMeta();
+                    giftMeta.setDisplayName(ChatColor.GOLD + "" + ChatColor.BOLD + "Gift Card");
+                    giftMeta.setLore(List.of(ChatColor.AQUA + "$" + helper.formatBigDecimal(amountInDecimal)));
+                    giftCard.setItemMeta(giftMeta);
+
+                    final ConfirmMenu confirmMenu = new ConfirmMenu.Builder(giftCard)
+                            .title(giftCard.getItemMeta().getDisplayName())
+                            .setOnConfirm(c -> {
+                                if (c) {
+                                    plugin.getServer().getDataCenter()
+                                            .runTransact(transaction -> {
+                                                WithdrawResult result = player.getSolarPlayer().getData(CreditsKey.INSTANCE)
+                                                        .withdrawBalance(transaction, amountInDecimal);
+
+                                                if (result.isSuccessful())
+                                                    createGiftCard(player, amount, giftCardCode -> player.sendMessage(ChatColor.translateAlternateColorCodes('&', "&aGift Card Code : &r&l&6" + giftCardCode)));
+                                                else player.sendMessage("Sorry, you don't have enough money!");
+                                            })
+                                            .exceptionally((ex) -> {
+                                                LOGGER.error("Failed to withdraw {} from {}", amount, player, ex);
+                                                return null;
+                                            });
+                                }
+                            })
+                            .build();
+                    confirmMenu.open(player);
+                } else
+                    sender.sendMessage("Sorry, but " + amountString + " is not a number!");
+            } else
+                sender.sendMessage("Please specify the Amount you want to use for the Gift Card");
+        } else
+            sender.sendMessage("Only Players use this Command");
+
+        return true;
+    }
+
+    /**
+     * amount Currency Value of the gift card
+     */
+    private void createGiftCard(CommandSender sender, double amount, Consumer<String> giftCardCode) {
```

Comment:
```suggestion
    private void createGiftCard(Player sender, double amount, Consumer<String> giftCardCode) {
```
So you can remove the useless cast